session.go

```diff
@@ -73,6 +69,8 @@ type session struct {
 	// closeChan is used to notify the run loop that it should terminate.
 	// If the value is not nil, the error is sent as a CONNECTION_CLOSE.
 	closeChan chan *qerr.QuicError
+	// the error this session was closed with
+	closeErr  error
```

Comment:
I see. Then I'd rather change the chan to a chan struct{} and always use the closeErr (and document it there). I really dislike having multiple spots for errors to propagate.